cashu/wallet/wallet.py

```diff
@@ -173,12 +228,20 @@ async def mint(self, amount: int, payment_hash: str = None):
         self.proofs += proofs
         return proofs
 
-    async def redeem(self, proofs: List[Proof]):
+    async def redeem(self, proofs: List[Proof], snd_secrets: List[str] = None):
+        if snd_secrets:
+            logger.debug(f"Redeption secrets: {snd_secrets}")
+            assert len(proofs) == len(snd_secrets)
+            # overload proofs with custom secrets for redemption
+            for p, s in zip(proofs, snd_secrets):
+                p.secret = s

```

Comment:
Upon redemption ("receive") of tokens with custom secret (defined by `cashu receive <token> -s <secret>`), custom secrets are inserted into proofs and then redeemed with the mint.